pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -451,6 +463,33 @@ func getGCPImageID(masterMachine *machineapi.Machine, scheme *runtime.Scheme, lo
 	return imageID, nil
 }
 
+// getNetwork retrieves the network information (Network name and subnet)
+// from existing machines on the remote cluster.
+func getNetwork(remoteMachineSets []machineapi.MachineSet,
+	scheme *runtime.Scheme, logger log.FieldLogger) (string, string, error) {
+	if len(remoteMachineSets) == 0 {
+		return "", "", nil
+	}
+
+	remoteMachineProviderSpec, err := decodeGCPMachineProviderSpec(
+		remoteMachineSets[0].Spec.Template.Spec.ProviderSpec.Value,
+		scheme,
+	)
+	if err != nil {
+		logger.WithError(err).Warn("cannot decode GCPMachineProviderSpec from master machine")
+		return "", "", errors.Wrap(err, "cannot decode GCPMachineProviderSpec from master machine")
+	}
+
+	if len(remoteMachineProviderSpec.NetworkInterfaces) == 0 {
+		logger.Warn("remove machine does not have any netowrk interfaces")
+		return "", "", nil
```

Comment:
should this be an error? vs currently no op?

Reply:
Probably an error.

Reply:
Fixed.